### ISSUE TITLE
Gatekeeper Warnings for DMG File on Mac

### ISSUE THREAD

**Issue by cupppcakes** [Chatterino/chatterino2#4636]
### Checklist

- [X] I'm reporting a problem with Chatterino
- [X] I've verified that I'm running **the most** recent nightly build or stable release
- [X] I've looked for my problem on the [wiki](https://wiki.chatterino.com/Help/)
- [X] I've searched the [issues and pull requests](https://github.com/Chatterino/chatterino2/issues?q=) for similar looking reports

### Describe your issue

I'm raising this issue in regards to the Chatterino _DMG image_ on macOS (I know there are older discussions and information about the Gatekeeper warnings and signing the .app file, but I believe this is the first regarding the .dmg that houses it).

As of Chatterino 2.4.3, it looks like macOS builds are now code signed, however with a self-signed certificate rather than an Apple-issued one. Furthermore, the enclosing DMG image is also being signed with this certificate. This is causing macOS to trigger a Gatekeeper warning when trying to open the .dmg in the first place, and then a second when trying to launch the Chatterino app itself.

Both can be bypassed by right-clicking the files and choosing Open from the context-menu, however this is actually a step back compared to before 2.4.3, when only the app (and not the DMG) would trigger a warning.

According to the codesign CLI utility, it's signed by:
```
Authority=chatterino-self-signed-4
Authority=pajlada’s CA 3
```
I'm assuming the use of self-signed certs are intentional here. (If you did happen to have actual Developer ID certs then they're not being applied to the published builds.)

### Screenshots

First Warning for Opening Chatterino.dmg (these were not shown prior to 2.4.3):
![Screenshot 2023-05-15 at 06 57 30](https://github.com/Chatterino/chatterino2/assets/129288938/70f80999-0a27-42cf-a562-433a9f9f7af8)
![Screenshot 2023-05-15 at 06 58 12](https://github.com/Chatterino/chatterino2/assets/129288938/e6c011fd-b08a-4358-92ff-359cb8b12f21)

Second Warning For Opening chatterino.app after allowing DMG:
![Screenshot 2023-05-15 at 06 58 36](https://github.com/Chatterino/chatterino2/assets/129288938/d0ce535a-da47-4523-882d-e88557ed9ad2)
![Screenshot 2023-05-15 at 06 59 05](https://github.com/Chatterino/chatterino2/assets/129288938/2e856322-0622-416f-916c-3266981b05f8)

### OS and Chatterino Version

Chatterino 2.4.4 (commit 29a14627) built with Qt 6.5.0 Running on macOS Ventura (13.3), kernel: 22.4.0

**Comment by cupppcakes** [comment]
Two ideas are:
1. Is is possible to leave the outer DMG image unsigned, so at least that still opens without triggering a warning? This seems like the quickest fix to get the level of friction presented to the user back to pre-2.4.3 levels.
2. Is it possible to obtain an Apple-issued signing certificate so Gatekeeper issues will no longer be a thing at all? This would be the ideal situation and would put Chatterino on the same footing as most other applications. I realize this costs money to apply for though (but I'm happy to help chip in some extra donations if it would make this possible).

**Comment by pajlada** [comment]
You're right, I'm using a self-signed certificate in the packaging process.

1. I can try not signing the dmg file for future releases, assuming that works that should be perfectly fine

2. The plan is to get an Apple-issued signing certificate, but they don't accept my Swedish passport so I'm in a bit of a pickle right now - looking into alternatives

**Comment by SputNikPlop** [comment]
I took a look at this on my personal account and I was able to get it to not display the warnings, also have a company one which I think we could use to sign the app. This also would let us distribute the app via the app store.

**Comment by ashugg** [comment]
Hi pajlada, is the issue that you are being blocked from joining the Apple Developer Program?

**Comment by pajlada** [comment]
> Hi pajlada, is the issue that you are being blocked from joining the Apple Developer Program?

Yup

**Comment by 8thony** [comment]
From https://github.com/Chatterino/chatterino2/issues/6454#issuecomment-3263879137
> I've signed up for the apple developer program now, hopefully 2.5.4 can undisable/undeprecate it

2.5.4 should only show the warning "is an app downloaded from the internet" and no other warning.

**Comment by pajlada** [comment]
Fixed in v2.5.4 - thanks everyone for the help reporting & testing. Nightly releases are codesigned but not notarized, so if you use those then you'll still get this warning. That is expected.

I am considering maybe spinning up our own brew tap with nightly or weekly releases where I do notarize them, but the flow of notarizing every single commit that flows into main seems unsustainable even if automated since there are notarization limits.
